addons/portal/static/src/js/portal_security.js

```diff
@@ -0,0 +1,266 @@
+        // in order to use `handleCheckIdentity`.
+        await handleCheckIdentity(this.proxy('_rpc'), this._rpc({
+            model: 'res.users',
+            method: 'api_key_wizard',
+            args: [session.user_id],
+        }));
+        const self = this;
+        const d_description = new Dialog(self, {
+            title: _t('New API Key'),
+            $content: qweb.render('portal.keydescription'),
+            buttons: [{text: _t('Confirm'), classes: 'btn-primary', close: true, click: async () => {
+                var description = d_description.el.querySelector('[name="description"]').value;
+                var wizard_id = await this._rpc({
+                    model: 'res.users.apikeys.description',
+                    method: 'create',
+                    args: [{name: description}],
+                });
+                var res = await handleCheckIdentity(
+                    this.proxy('_rpc'),
+                    this._rpc({
+                        model: 'res.users.apikeys.description',
+                        method: 'make_key',
+                        args: [wizard_id],
+                    })
+                );
+                const d_show = new Dialog(self, {
+                    title: _t('API Key Ready'),
+                    $content: qweb.render('portal.keyshow', {key: res.context.default_key}),
+                    buttons: [{text: _t('Close'), clases: 'btn-primary', close: true}],
+                });
+                d_show.on('closed', this, () => {
+                    window.location = window.location;
+                });
+                d_show.open();
+            }}, {text: _t('Discard'), close: true}],
+        });
+        d_description.opened(() => {
+            const input = d_description.el.querySelector('[name="description"]');
+            input.focus();
+            d_description.el.addEventListener('submit', (e) => {
+                e.preventDefault();
+                d_description.$footer.find('.btn-primary').click();
+            });
+        });
+        d_description.open();
+    }
+});
+
+publicWidget.registry.RemoveAPIKeyButton = publicWidget.Widget.extend({
+    selector: '.o_portal_remove_api_key',
+    events: {
+        click: '_onClick'
+    },
+
+    async _onClick(e){
+        e.preventDefault();
+        await handleCheckIdentity(
+            this.proxy('_rpc'),
+            this._rpc({
+                model: 'res.users.apikeys',
+                method: 'remove',
+                args: [parseInt(this.el.id)]
+            })
+        );
+        window.location = window.location;
+    }
+});
+
+publicWidget.registry.portalSecurity = publicWidget.Widget.extend({
+    selector: '.o_portal_security_body',
+
+    /**
+     * @override
+     */
+    init: function () {
+        // Show the "deactivate your account" modal if needed
+        $('.modal.show#portal_deactivate_account_modal').removeClass('d-block').modal('show');
```

Comment:
Avoid using jquery, we plan to remove it :) use classList.add for example (Edit: I see you already use it in this file, just re-read everything once to avoid jquery addition :) )